ubench.h

```diff
@@ -306,18 +319,19 @@ UBENCH_EXTERN struct ubench_state_s ubench_state;
 #pragma clang diagnostic pop
 #endif
 
-#define UBENCH(SET, NAME)                                                      \
+static UBENCH_INLINE int ubench_keep_running(struct ubench_run_state_s* ubs)
+{
+  ubench_int64_t curr_sample = ubs->sample++;
+  ubs->ns[curr_sample] = ubench_ns();
+  return curr_sample < ubs->size ? 1 : 0;
+}
+
+#define UBENCH_KEEP_RUNNING()                                                  \
```

Comment:
I do wonder if there is anyway for us to ban calling this within a `UBENCH` (EG. not the `_EX` variant).

Reply:
I don't think so, but I'll have a think about it!